File: Monolitico/src/main/java/br/tcc/monolitico/controller/ProdutoController.java
package br.tcc.monolitico.controller;

import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import br.tcc.monolitico.record.ProdutoRecord;
import br.tcc.monolitico.service.ProdutoService;
import jakarta.validation.Valid;

@RestController
@RequestMapping("/produto")
public class ProdutoController {

	public final ProdutoService produtoService;
	
	public ProdutoController(ProdutoService produtoService) {
		this.produtoService = produtoService;
	}
	
	@GetMapping("{id}")
	public ResponseEntity<Object> getById(@PathVariable final Long id) {
		return produtoService.getById(id);
	}
	
	@GetMapping
	public ResponseEntity<Object> get(@PageableDefault(page = 0, size = 10) final Pageable page) {
		return produtoService.get(page);
	}
	
	@DeleteMapping("{id}")
	public ResponseEntity<Object> delete(@PathVariable final Long id) {
		return produtoService.delete(id);
	}

	@PutMapping("{id}")
	public ResponseEntity<Object> update(@PathVariable final Long id, @RequestBody @Valid final ProdutoRecord dto) {
		return produtoService.update(id, dto);
	}
	
	@PostMapping
	public ResponseEntity<Object> save(@RequestBody @Valid final ProdutoRecord dto) {
		return produtoService.save(dto);
	}

}

File: Python/produto/src/controller/produto_controller.py
from fastapi import APIRouter
from service import produto_service
from dto.produto_DTO import ProdutoDTO;

router = APIRouter()

@router.get("/valorBaixo")
def get_all_by_valor_baixo():
    return produto_service.get_all_by_valor_baixo()

@router.get("/{id}")
def get_by_id(id: int):
	return produto_service.get_by_id(id)

@router.get("")
def get(page: int = 0, size: int = 10, sort: str = "id,ASC"):
	return produto_service.get(page, size, sort)

@router.delete("/{id}")
def delete(id: int):
	return produto_service.delete(id)

@router.put("/{id}")
def update(id: int, produtoDTO: ProdutoDTO):
	return produto_service.update(id, produtoDTO)

@router.post("")
def save(produtoDTO: ProdutoDTO):
	return produto_service.save(produtoDTO)